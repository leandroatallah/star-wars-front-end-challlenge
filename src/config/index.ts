export const CHANGE_THEME = 'CHANGE_THEME'
export const THEME_DARK_SIDE = 'THEME_DARK_SIDE'
export const THEME_LIGHT_SIDE = 'THEME_LIGHT_SIDE'
