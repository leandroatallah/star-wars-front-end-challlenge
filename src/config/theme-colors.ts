type myTheme = {
    colors: {
        blue: string
        yellow: string
        black: string
        white: string
    }
}

const themeColors: myTheme = {
    colors: {
        blue: '#2C97D1',
        yellow: '#FBFE63',
        black: '#2A2A2A',
        white: '#FFF',
    },
}

export default themeColors