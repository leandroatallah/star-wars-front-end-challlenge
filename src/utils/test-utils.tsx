import { Provider } from 'react-redux'
import { render } from '@testing-library/react'
import { CHANGE_THEME } from '../config'
import { createStore, Store } from 'redux'

const startingState = { theme: '' }
function reducer(state = startingState, action: { type: string; payload: string }) {
    switch (action.type) {
        case CHANGE_THEME:
            return { ...state, theme: action.payload }
        default:
            return state
    }
}

type storeProps = {
    initialState?: {
        theme: string
    },
    store?: Store<{
        theme: string;
    }, {
        type: string;
        payload: string;
    }>
}

export function renderWithRedux(
    component: JSX.Element,
    { initialState, store = createStore(reducer, initialState) }: storeProps = {},
) {
    return {
        ...render(<Provider store={store} > {component} </Provider>),
    }
}
