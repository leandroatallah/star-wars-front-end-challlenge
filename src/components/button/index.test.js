import '@testing-library/jest-dom'
import 'jest-styled-components'
import { ThemeProvider } from 'styled-components'
import Button from '.'
import themeColors from '../../config/theme-colors'
import { THEME_DARK_SIDE, THEME_LIGHT_SIDE } from '../../config'
import { cleanup } from '@testing-library/react'
import { createStore } from 'redux'
import { renderWithRedux } from '../../utils/test-utils'

afterEach(cleanup)

describe('<Button />', () => {
    it('Render with redux', () => {
        const { getByTestId } = renderWithRedux(
            <ThemeProvider theme={themeColors} >
                <Button>Button Text </Button>)
            </ThemeProvider>,
        )
        expect(getByTestId('button')).toMatchSnapshot()
    })

    it('should have blue background and white color by default', () => {
        const { getByTestId } = renderWithRedux(
            <ThemeProvider theme={themeColors} >
                <Button>Button Text </Button>)
            </ThemeProvider>,
        )
        expect(getByTestId('button')).toHaveStyleRule('background-color', themeColors.colors.blue)
        expect(getByTestId('button')).toHaveStyleRule('color', themeColors.colors.white)
    })

    it('should have black background and yellow color when theme is set to light', () => {
        const store = createStore(() => ({ theme: THEME_LIGHT_SIDE }))
        const { getByTestId } = renderWithRedux(
            <ThemeProvider theme={themeColors} >
                <Button>Button Text </Button>)
            </ThemeProvider>,
            { store },
        )
        expect(getByTestId('button')).toHaveStyleRule('background-color', themeColors.colors.black)
        expect(getByTestId('button')).toHaveStyleRule('color', themeColors.colors.yellow)
    })

    it('should have white background and black color when theme is set to dark', () => {
        const store = createStore(() => ({ theme: THEME_DARK_SIDE }))
        const { getByTestId } = renderWithRedux(
            <ThemeProvider theme={themeColors} >
                <Button>Button Text </Button>)
            </ThemeProvider>,
            { store },
        )
        expect(getByTestId('button')).toHaveStyleRule('background-color', themeColors.colors.white)
        expect(getByTestId('button')).toHaveStyleRule('color', themeColors.colors.black)
    })
})
