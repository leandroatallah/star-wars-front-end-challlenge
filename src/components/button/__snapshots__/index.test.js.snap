// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Render with redux 1`] = `
.c0 {
  background-color: #2C97D1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 56px;
  font-size: 18px;
  line-height: 1.218;
  font-weight: 700;
  padding: 0 37px;
  border-radius: 10px;
  color: #FFF;
  cursor: pointer;
  pointer-events: auto;
  -webkit-transition: all .2s ease-out;
  transition: all .2s ease-out;
}

.c0:hover {
  opacity: 0.8;
}

.c0 svg {
  display: none;
  width: 20px;
  height: 20px;
  margin: 0;
  margin-left: 10px;
  vertical-align: middle;
}

<button
  class="c0"
  data-testid="button"
  loading="0"
  type="button"
>
  <span>
    Button Text 
  </span>
  <svg
    preserveAspectRatio="xMidYMid"
    viewBox="0 0 100 100"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink="http://www.w3.org/1999/xlink"
  >
    <circle
      cx="50"
      cy="50"
      fill="none"
      r="35"
      stroke="#ffffff"
      stroke-dasharray="164.93361431346415 56.97787143782138"
      stroke-width="10"
    >
      <animatetransform
        attributeName="transform"
        dur="1s"
        keyTimes="0;1"
        repeatCount="indefinite"
        type="rotate"
        values="0 50 50;360 50 50"
      />
    </circle>
  </svg>
</button>
`;
